web/src/emoji_picker.js

```diff
@@ -699,34 +666,72 @@ export function toggle_emoji_popover(target, id, additional_popover_options) {
         },
     );
 }
-function register_click_handlers() {
-    $(document).on("click", ".emoji-popover-emoji.reaction", function (e) {
-        // When an emoji is clicked in the popover,
-        // if the user has reacted to this message with this emoji
-        // the reaction is removed
-        // otherwise, the reaction is added
-        const emoji_name = $(this).attr("data-emoji-name");
-        toggle_reaction(emoji_name, e);
-    });
 
-    $(document).on("click", ".emoji-popover-emoji.composition", function (e) {
-        const emoji_name = $(this).attr("data-emoji-name");
-        const emoji_text = ":" + emoji_name + ":";
-        // The following check will return false if emoji was not selected in
-        // message edit form.
-        if (edit_message_id !== null) {
-            const $edit_message_textarea = $(
-                `#edit_form_${CSS.escape(edit_message_id)} .message_edit_content`,
-            );
-            // Assign null to edit_message_id so that the selection of emoji in new
-            // message composition form works correctly.
-            edit_message_id = null;
-            compose_ui.insert_syntax_and_focus(emoji_text, $edit_message_textarea);
-        } else {
-            compose_ui.insert_syntax_and_focus(emoji_text);
+function handle_reaction_emoji_clicked(emoji_name, event) {
+    // When an emoji is clicked in the popover,
+    // if the user has reacted to this message with this emoji
+    // the reaction is removed
+    // otherwise, the reaction is added
+    toggle_reaction(emoji_name, event);
+}
+
+function handle_status_emoji_clicked(emoji_name) {
+    hide_emoji_popover();
+    let emoji_info = {
+        emoji_name,
+        emoji_alt_code: user_settings.emojiset === "text",
+    };
+    if (!emoji_info.emoji_alt_code) {
+        emoji_info = {...emoji_info, ...emoji.get_emoji_details_by_name(emoji_name)};
+    }
+    user_status_ui.set_selected_emoji_info(emoji_info);
+    user_status_ui.update_button();
+    user_status_ui.toggle_clear_message_button();
+}
+
+function handle_composition_emoji_clicked(emoji_name) {
+    hide_emoji_popover();
+    const emoji_text = ":" + emoji_name + ":";
+    // The following check will return false if emoji was not selected in
+    // message edit form.
+    if (edit_message_id !== null) {
+        const $edit_message_textarea = $(
+            `#edit_form_${CSS.escape(edit_message_id)} .message_edit_content`,
+        );
+        // Assign null to edit_message_id so that the selection of emoji in new
+        // message composition form works correctly.
+        edit_message_id = null;
+        compose_ui.insert_syntax_and_focus(emoji_text, $edit_message_textarea);
+    } else {
+        compose_ui.insert_syntax_and_focus(emoji_text);
+    }
+}
+
+function handle_emoji_clicked($emoji, event) {
+    const emoji_name = $emoji.attr("data-emoji-name");
+    const emoji_type = $emoji.attr("data-emoji-type");
```

Comment:
This looks great except that we really don't need a `data-emoji-type` on every single emoji element in the picker, and that name is confusing given it is not equal to `emoji.type`.

I'm pretty sure we don't allow multiple emoji picker popovers open at the same time, so simplest would just be to store this value in a global variable or an attribute on the popover as a whole. As a global variable, I'd call it something like `emoji_destination`.

Reply:
Renamed the data attribute to `data-emoji-destination` and shifted it to `.emoji-picker-popover` div as a global variable.